backend/src/main/java/de/neuefische/booklibrary/backend/api/ApiBookService.java

```diff
@@ -0,0 +1,107 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class ApiBookService {
+    private static final String PATH_QUERY = "?q=";
+    private static final String PATH_KEY = "&key=";
+    public static final String DEFAULT_COVER = "https://simg.nicepng.com/png/small/251-2515797_no-circle-book-book-logo-png.png";
+    private final String apiKey;
+    private final WebClient webClient;
+
+    public ApiBookService(@Value("${api.url}") String apiUrl, @Value("${api.key}") String apiKey) {
+        this.apiKey = apiKey;
+        this.webClient = WebClient.create(apiUrl);
+    }
+
+    public List<Book> getApiBookByIsbn(String isbn) throws ResponseStatusException {
+        String searchByIsbn = "isbn:";
+        String query = PATH_QUERY + searchByIsbn + isbn + PATH_KEY + apiKey;
+        ResponseEntity<BookResponseElement> bookResponse = getBookResponse(query);
+        List<ApiBook> books = getBookList(bookResponse).stream().filter(book -> book.volumeInfo()
+                .industryIdentifiers()
+                .stream()
+                .anyMatch(isbn1 -> isbn.equals(isbn1.identifier()))).toList();
+        List<Book> bookList = new ArrayList<>();
+        getBookListFromApiBookList(books, bookList);
+        return bookList;
+
+    }
+
+    private static void getBookListFromApiBookList(List<ApiBook> books, List<Book> bookList) {
+        for (ApiBook apiBook : books) {
+            RentBookInfo rentBookInfo = new RentBookInfo("", "");
+            VolumeInfo volumeInfo = apiBook.volumeInfo();
+            String thumbnail = Optional.ofNullable(volumeInfo.imageLink())
+                    .map(ImageLink::thumbnail)
+                    .orElse(DEFAULT_COVER);
+            String author = Optional.ofNullable(volumeInfo.authors()).map(current -> current.get(0))
+                    .orElse(null);
+
+            String category = Optional.ofNullable(volumeInfo.categories()).map(current -> current.get(0))
+                    .orElse(null);
+            int pageCount = Optional.of(volumeInfo.pageCount()).orElse(null);
+            String printType = volumeInfo.printType();
+
+            Book book = new Book(
+                    apiBook.id(),
+                    thumbnail,
+                    apiBook.volumeInfo().title(),
+                    author,
+                    apiBook.volumeInfo().industryIdentifiers(),
+                    category,
+                    printType,
+                    pageCount,
+                    Availability.AVAILABLE, rentBookInfo);
```

Comment:
Ein Zeilenumbruch wäre hier konsequent.
```suggestion
                    Availability.AVAILABLE,
                    rentBookInfo);
```

Reply:
Ein Zeilenumbruch hinzugefügt